nix-eval-jobs doesn't build @Mic92 @adisbladis 

```log
nix-eval-jobs> unpacking sources
nix-eval-jobs> unpacking source archive /nix/store/wndp1lpb7g9gs3fmh4pjmqw80pjiw606-source
nix-eval-jobs> source root is source
nix-eval-jobs> patching sources
nix-eval-jobs> configuring
nix-eval-jobs> meson flags: --buildtype=plain         --libdir=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1/lib --libexecdir=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1/libexec         --bindir=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1/bin --sbindir=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1/sbin         --includedir=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1/include         --mandir=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1/share/man--infodir=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1/share/info         --localedir=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1/share/locale         -Dauto_features=enabled         -Dwrap_mode=nodownload         --prefix=/nix/store/pa8kr05sf59s131vpi1qajp8yxzsxlhl-nix-eval-jobs-0.0.1
nix-eval-jobs> The Meson build system
nix-eval-jobs> Version: 0.57.1
nix-eval-jobs> Source dir: /build/source
nix-eval-jobs> Build dir: /build/source/build
nix-eval-jobs> Build type: native build
nix-eval-jobs> Project name: nix-eval-jobs
nix-eval-jobs> Project version: 0.1.6
nix-eval-jobs> C++ compiler for the host machine: g++ (gcc 10.3.0 "g++ (GCC) 10.3.0")
nix-eval-jobs> C++ linker for the host machine: g++ ld.bfd 2.35.2
nix-eval-jobs> Host machine cpu family: x86_64
nix-eval-jobs> Host machine cpu: x86_64
nix-eval-jobs> Found pkg-config: /nix/store/zm7d9fw403racbfyprb4hqlwbwz8544n-pkg-config-wrapper-0.29.2/bin/pkg-config (0.29.2)
nix-eval-jobs> Run-time dependency nix-main found: YES 2.5pre20211125_1d2dbbb
nix-eval-jobs> Run-time dependency nix-store found: YES 2.5pre20211125_1d2dbbb
nix-eval-jobs> Run-time dependency nix-expr found: YES 2.5pre20211125_1d2dbbb
nix-eval-jobs> Run-time dependency threads found: YES
nix-eval-jobs> Run-time dependency nlohmann_json found: YES 3.10.2
nix-eval-jobs> Run-time dependency Boost found: YES 1.69.0 (/nix/store/v14bqrnwvgs7r8x5x6l55gajplzlb96p-boost-1.69.0-dev)
nix-eval-jobs> Build targets in project: 1
nix-eval-jobs> Found ninja-1.10.2 at /nix/store/p8x0kw8vb5m74ddm7vzpan7hmwwflbsm-ninja-1.10.2/bin/ninja
nix-eval-jobs> meson: enabled parallel building
nix-eval-jobs> building
nix-eval-jobs> build flags: -j22 -l22
nix-eval-jobs> [1/2] Compiling C++ object src/nix-eval-jobs.p/nix-eval-jobs.cc.o
nix-eval-jobs> FAILED: src/nix-eval-jobs.p/nix-eval-jobs.cc.o
nix-eval-jobs> g++ -Isrc/nix-eval-jobs.p -Isrc -I../src -I/nix/store/za2c4sf50k1kf1yadmwhk9cpjn9iixpf-nix-2.5pre20211125_1d2dbbb-dev/include/nix -I/nix/store/76mpr3rwfbbijmjnwkk3bcajfkqanrjb-boehm-gc-8.0.6-dev/include -I/nix/store/v14bqrnwvgs7r8x5x6l55gajplzlb96p-boost-1.69.0-dev/include -I/nix/store/xqrdd2vahf1mi2ww0dwbs181x8rkpxbd-nlohmann_json-3.10.2/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -pthread -DBOOST_ALL_NO_LIB -std=c++17 -std=c++17 -std=c++17 -std=c++17 -fvisibility=hidden -MD -MQ src/nix-eval-jobs.p/nix-eval-jobs.cc.o -MF src/nix-eval-jobs.p/nix-eval-jobs.cc.o.d -o src/nix-eval-jobs.p/nix-eval-jobs.cc.o -c ../src/nix-eval-jobs.cc
nix-eval-jobs> ../src/nix-eval-jobs.cc:32:8: warning: base class 'struct nix::MixEvalArgs' has accessible non-virtual destructor [-Wnon-virtual-dtor]
nix-eval-jobs>    32 | struct MyArgs : MixEvalArgs, MixCommonArgs
nix-eval-jobs>       |        ^~~~~~
nix-eval-jobs> ../src/nix-eval-jobs.cc:32:8: warning: base class 'class nix::MixCommonArgs' has accessible non-virtual destructor [-Wnon-virtual-dtor]
nix-eval-jobs> ../src/nix-eval-jobs.cc:32:8: warning: 'struct MyArgs' has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
nix-eval-jobs> ../src/nix-eval-jobs.cc: In function 'void worker(nix::EvalState&, nix::Bindings&, nix::AutoCloseFD&, nix::AutoCloseFD&, const Path&)':
nix-eval-jobs> ../src/nix-eval-jobs.cc:191:72: error: no matching function for call to 'printValueAsJSON(nix::EvalState&, bool, nix::Value&,std::stringstream&, nix::PathSet&)'
nix-eval-jobs>   191 |                   printValueAsJSON(state, true, *metaValue, ss, context);
nix-eval-jobs>       |                                                                        ^
nix-eval-jobs> In file included from ../src/nix-eval-jobs.cc:20:
nix-eval-jobs> /nix/store/za2c4sf50k1kf1yadmwhk9cpjn9iixpf-nix-2.5pre20211125_1d2dbbb-dev/include/nix/value-to-json.hh:13:6: note: candidate: 'void nix::printValueAsJSON(nix::EvalState&, bool, nix::Value&, const nix::Pos&, nix::JSONPlaceholder&, nix::PathSet&)'
nix-eval-jobs>    13 | void printValueAsJSON(EvalState & state, bool strict,
nix-eval-jobs>       |      ^~~~~~~~~~~~~~~~
nix-eval-jobs> /nix/store/za2c4sf50k1kf1yadmwhk9cpjn9iixpf-nix-2.5pre20211125_1d2dbbb-dev/include/nix/value-to-json.hh:13:6: note:   candidate expects 6 arguments, 5 provided
nix-eval-jobs> /nix/store/za2c4sf50k1kf1yadmwhk9cpjn9iixpf-nix-2.5pre20211125_1d2dbbb-dev/include/nix/value-to-json.hh:16:6: note: candidate: 'void nix::printValueAsJSON(nix::EvalState&, bool, nix::Value&, const nix::Pos&, std::ostream&, nix::PathSet&)'
nix-eval-jobs>    16 | void printValueAsJSON(EvalState & state, bool strict,
nix-eval-jobs>       |      ^~~~~~~~~~~~~~~~
nix-eval-jobs> /nix/store/za2c4sf50k1kf1yadmwhk9cpjn9iixpf-nix-2.5pre20211125_1d2dbbb-dev/include/nix/value-to-json.hh:16:6: note:   candidate expects 6 arguments, 5 provided
nix-eval-jobs> ninja: build stopped: subcommand failed.
```

--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -256,13 +256,13 @@ in rec {
   nixUnstable = lib.lowPrio (callPackage common rec {
     pname = "nix";
     version = "2.5${suffix}";
-    suffix = "pre20211007_${lib.substring 0 7 src.rev}";
+    suffix = "pre20211126_${lib.substring 0 7 src.rev}";
 
     src = fetchFromGitHub {
       owner = "NixOS";
       repo = "nix";
-      rev = "844dd901a7debe8b03ec93a7f717b6c4038dc572";
-      sha256 = "sha256-fe1B4lXkS6/UfpO0rJHwLC06zhOPrdSh4s9PmQ1JgPo=";
+      rev = "55275fcc5966cfad80fb6dc77b8d8939a2f1b8e0";
+      sha256 = "sha256-jr85Z+0TA86mdsi3Y91wM3dFHWemmFHKdNwbx9rGz2U=";
     };
 
     boehmgc = boehmgc_nixUnstable;

